ui/app/scripts/controllers/create.js

```diff
@@ -132,12 +132,17 @@
         statusChanged: function() {
           model.demo.demoStatus.lastStatusTimestamp = new Date().toJSON();
         },
-        deleteAttachment: function(attachent, index) {
-          demoService.deleteAttachment($routeParams.uri, attachent).then(
+        deleteAttachment: function(attachment, index) {
+          demoService.deleteAttachment($routeParams.uri, attachment).then(
             function() {
               model.demo.attachments.splice(index, 1);
             }
           );
+          for (var i = 0; i < model.demo.memos.length; i++) {
+              if (model.demo.memos[i].title === attachment.attachmentName) {
+                model.demo.memos.splice(i, 1);
+              }
+          }
```

Comment:
Only when Submit is clicked. I hesitated to add a save after the memos were deleted since that would also save any changes the user before the Submit button was clicked.

Inserting the code for deleting the memos in a post commit trigger results in a trigger storm. There should be a better place for the logic but I think it means a bigger implementation overall.